File: comments.py
# |Linear Axis MSP430 Example|-------------------------------------------------
#
# Project: Linear Axis MSP430 Example
# Program: linear_axis.py
#
# Description:
#   This program is designed to manipulate the timer output of the MSP430. The
# signals are tailored for Teknic's MCPV ClearPath Motor. The simplest controls
# required are pure digital signals. Other inputs include frequency and PWM
# commands. Frequency commands are preferred as the bi-directional PWM signal 
# requires a deadband around 50% duty to ensure the motor stops. The outputs 
# for each of the control modes will be sent to separate pins to avoid 
# unintentional motion commands being being received by the motor.
#
#   This purpose of this project is to demonstrate the use of the linear axis
# in a lab setting at the British Columbia Institute of Technology. The
# secondary purpose for this project is to be used a Co-Op work term report for
# the University of Victoria.
#
# The program is modified for the Raspberry Pi.
#
# Author: Isaiah Regacho
# 
# Last Modified: October 23, 2021
# -----------------------------------------------------------------------------

# |Package Version Control|----------------------------------------------------
# Python 3.9.2 (default on Raspberry Pi)
#
# Package              | Version    : Command
# ==================== | ========== : =========================================
# matplotlib           | 3.4.3      : pip install matplotlib
# numpy                | 1.21.3     : pip install --upgrade numpy
# pandas               | 1.3.4      : pip install pandas
# pyserial             | 3.5b0      : pip install pyserial
# pyusb                | 1.2.1      : pip install pyusb (requires libusb)
# ==================== | ========== : =========================================
# -----------------------------------------------------------------------------

# |MODULES|--------------------------------------------------------------------


    # -------------------------------------------------------------------------
    # Method: reset_count
    #
    # Description:
    #       This method resets the HLFB counter.
    #
    # Last Modified: March 18, 2021 by Isaiah Regacho
    # -------------------------------------------------------------------------

File: comments.c
/*|Name of the Project|--------------------------------------------------------
#
# Project: Name of the Project (NOP)
# Program: comments.c
#
# Description:
#   This file is used as a template for Isaiah's style of commenting code. This
# paragraph should contain a description of the project or program. The main 
# file should include important details for future reference.
#
# Peripherals:
#   MCPV-3210S-RQN  - Teknic ClearPath Digital I/O Series for Position Control
#   Python          - UART/Serial Python GUI
#   <Peripheral>    - <short descriptions only>
#
# Signals/Pins:
#   Input A         P3.0    Digital Output
#   Input B         P3.1    Digital Output
#   Enable Input    P3.2    Digital Output
#   HLFB            P1.3    Timer Input
#   <Signal>        <Pin>   <Optional Description>
#
# Author: <Name>
# Date Created: <Date>
# Last Modified: <Today>
# 
# Note: Make it a habit to update the date each day you modify your code.
# -----------------------------------------------------------------------------*/

/*|Includes|-------------------------------------------------------------------*/
/*
#include <example.h>    // <list of functions used> - <purpose>
#include <string.h>     // strtok - Command parsing.
*/

/*|Globals|--------------------------------------------------------------------*/

/*|Macros|---------------------------------------------------------------------*/
#define set_bit(PORT, PIN, BIT) PORT = (BIT) ? (PORT | PIN) : (PORT &~ PIN)

/*|Function Declarations|------------------------------------------------------*/
void initClocks(void);
void executeAction(enum action, int);
void sendData(void);

/*|Main|-----------------------------------------------------------------------*/
int main(void){
	WDTCTL = WDTPW | WDTHOLD;	// Stop the Watchdog Timer
	
	// Variables

	// Initialize MCLK = SMCLK = 8.192 MHz
	initClocks();

	// Initialize GPIO
	initGPIO();

	// Initialize UART Serial Communication
	initUART();

	// Initialize Timer A0 for PWM or Frequency Control
	initTimerA0();

	// Enable Global Interrupts
	__enable_interrupt();

	// Main Loop
	while(42) {

	}

	return 0;
}

/*|functionName|---------------------------------------------------------------
#
# Description:
#   This function... <Description>
#
# Inputs:
#   a   - a number
#   b   - b number
#
# Returns: void
#
# Last Modified: March 1, 2022 by Isaiah Regacho
# -----------------------------------------------------------------------------*/
void functionName(enum action a, int b) {
    for(;;);
}
